### PR TITLE
feat(instance_snapshot): add snapshot import

### LINKED ISSUE #1458: Import of snapshot
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Snapshots can now be imported from qcow2
https://developers.scaleway.com/en/products/instance/api/#post-f271c8
It would be nice to be able to create an instance from a qcow file

### New or Affected Resource(s)

* scaleway_instance_snapshot

### Potential Terraform Configuration

```hcl

provider "scaleway" {
  zone = "fr-par-2"
}

data scaleway_object_bucket bucket {
  name = "mybucket"
}

resource scaleway_object "qcow" {
  bucket = data.scaleway_object_bucket.bucket
  name = "image.qcow2"

  source = "image.qcow2"
  hash = filemd5("image.qcow2")
}

resource "scaleway_instance_snapshot" "snapshot" {
  type = "unified"
  bucket = data.scaleway_object_bucket.bucket.name
  key = scaleway_object.qcow.name
}

resource "scaleway_instance_volume" "volume" {
  type = "b_ssd"
  from_snapshot_id = scaleway_instance_snapshot.snapshot.id
}

resource scaleway_instance_image "image" {
  root_volume_id = scaleway_instance_volume.volume.id
}

resource scaleway_instance_server "server" {
  image = scaleway_instance_image.image.id
  type  = "PLAY2-MICRO"
}
```

### References

* #1457

### PR DESCRIPTION
**Pull request by Codelax**
Close #1458

### CODE REVIEW

File: docs/resources/instance_snapshot.md [hunk 1]
```diff
@@ -59,6 +81,9 @@ Updates to this field will recreate a new resource.
 - `project_id` - (Defaults to [provider](../index.md#project_id) `project_id`) The ID of the project the snapshot is
   associated with.
 - `tags` - (Optional) A list of tags to apply to the snapshot.
+- `import` - (Optional) Import a snapshot from a qcow2 file located in a bucket
+  - `bucket` - Bucket name containing qcow2 to import
```
Comment (Monitob) [inline docs/resources/instance_snapshot.md:85]:
```suggestion
  - `bucket` - Bucket name containing [qcow2](https://en.wikipedia.org/wiki/Qcow) to import
```

File: scaleway/resource_instance_snapshot.go [hunk 2]
```diff
@@ -67,6 +67,27 @@ func resourceScalewayInstanceSnapshot() *schema.Resource {
...
+						},
+						"key": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Key of the qcow file in the specified bucket",
+						},
+					},
+				},
+				Optional:    true,
+				Description: "Import snapshot from a qcow",
```
Comment (Monitob) [inline scaleway/resource_instance_snapshot.go:89]: Should we add a conflictWith `volume_id`?

File: docs/resources/instance_snapshot.md (hunk 1 above)
Comment (Monitob) [inline docs/resources/instance_snapshot.md:85]: There is any link on the scaleway documentation website?

File: scaleway/resource_instance_snapshot.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ func resourceScalewayInstanceSnapshot() *schema.Resource {
 			},
 			"volume_id": {
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
```
Comment (Monitob) [inline scaleway/resource_instance_snapshot.go:39]: Is it possible to snap without volumeID?  As long as your example show yes, why is set on your request to create?

**Review by Monitob — CHANGES_REQUESTED** [review]

File: scaleway/resource_instance_snapshot.go (hunk 3 above)
Comment (Codelax) [reply scaleway/resource_instance_snapshot.go:39]: Import of qcow is made without `volume_id`

**Review by Monitob — APPROVED** [review]

**Pull request merged**